pkg/clusteragent/admission/mutate/common/common.go

```diff
@@ -11,29 +11,33 @@ package common
 import (
 	"encoding/json"
 	"fmt"
+	"strconv"
 
 	"github.com/wI2L/jsondiff"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/dynamic"
 
 	admCommon "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/metrics"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // MutationFunc is a function that mutates a pod
-type MutationFunc func(*corev1.Pod, string, dynamic.Interface) error
+type MutationFunc func(*corev1.Pod, string, dynamic.Interface) (bool, error)
 
 // Mutate handles mutating pods and encoding and decoding admission
 // requests and responses for the public mutate functions
-func Mutate(rawPod []byte, ns string, m MutationFunc, dc dynamic.Interface) ([]byte, error) {
+func Mutate(rawPod []byte, ns string, mutationType string, m MutationFunc, dc dynamic.Interface) ([]byte, error) {
 	var pod corev1.Pod
 	if err := json.Unmarshal(rawPod, &pod); err != nil {
 		return nil, fmt.Errorf("failed to decode raw object: %v", err)
 	}
 
-	if err := m(&pod, ns, dc); err != nil {
-		return nil, err
+	if injected, err := m(&pod, ns, dc); err != nil {
+		metrics.MutationAttempts.Inc(mutationType, metrics.StatusError, strconv.FormatBool(injected), err.Error())
+	} else {
+		metrics.MutationAttempts.Inc(mutationType, metrics.StatusSuccess, strconv.FormatBool(injected), "")
```

Comment:
I'm wondering if the `metrics.MutationAttempts` field should be created with [NewSimpleCounterWithOpts()](https://github.com/DataDog/datadog-agent/blob/e28da0ffc545969927a951d25dc48caaab383679/comp/core/telemetry/telemetry.go#L125C25-L125C50)` instead of a gauge. 
An `Simple` to not have a fix number of tags